### DISCUSSION TITLE
Path is mismatching?

### DISCUSSION THREAD

**Discussion by dalisoft** [uNetworking/uWebSockets.js#336] [Q&A] [answered]
Hi.
I found bug related to `uWS.App()` methods (get, post)

## Code

```js
import uWS from 'uWebSockets.js';

const app = uWS.App();

app.get('/', (res) => {
  res.end('hello world');
});

app.get('/foo/', (res) => {
  res.end('hello at /foo/');
});

app.get('/bar', (res) => {
  res.end('hello at /bar');
});

app.listen(
  4000,
  (token) => token && console.debug('server started at port', 4000)
);
```

## Test

1. Run the code above
2. Go-to browsers and try to run these urls

|URL|Passing|Error|
|-|-|-|
|`/foo`|No|Yes|
|`/foo/`|Yes|No|
|`/bar`|Yes|No|
|`/bar/`|No|Yes|

## Question

Is there possible to define path which works both with `/foo` and `/foo/` or `/bar` and `/bar/`?

Thanks for best project

**Comment by hst-m** [comment, marked as answer]
trailing slash is considered a different URL so it's not a bug, you could check for trailing slash in your handler and remove it or redirect to non-trailing slash url:
```javascript
// remove trailing slash before processing request
app.any('/*', (res, req) => {
  const method = req.getMethod()
  let path = req.getUrl()
  if(path.endsWith('/') && path != '/') path = path.slice(0,-1)
  switch(method) {
    case 'get': switch(path) {
      case '/': return res.end(`get ${path}`)
      case '/foo': return res.end(`get ${path}`)
      case '/bar': return res.end(`get ${path}`)
    } break
    case 'post': switch(path) {
      case '/': return res.end(`post ${path}`)
      case '/foo': return res.end(`post ${path}`)
    } break
  }
  res.writeStatus('404').end()
})
```
```javascript
// redirect to non-trailing slash url
app.any('/*', (res, req) => {
  const host = req.getHeader('host')
  const path = req.getUrl()
  const query = req.getQuery()
  if(path.endsWith('/') && path != '/') return res.writeStatus('301').writeHeader('Location',`https://${host}${path.slice(0,-1)}${query?'?'+query:''}`).end()
})
```

**Comment by dalisoft** [comment]
@hst-m Thanks for temporarily solution, but it worked before (not remember on which version).
But recently got new bug at [here](https://github.com/nanoexpress/pro/issues/41) and myself also faced this issue

**Comment by dalisoft** [comment]
I fixed it in my project new [release](https://github.com/nanoexpress/pro/releases/tag/v2.0.2) with [hot-fix temporarily solution](https://github.com/nanoexpress/pro/commit/5b24de2edaaca383df895ff2b8784659140eba98).

I'm think will be correct if this behavior fixed, but i may be wrong as i don't have enough experience in HTTP Specs.
Anyway thank you @hst-m for fast reply

**Comment by hst-m** [comment]
ok just to note, they are different urls can read here: https://webmasters.googleblog.com/2010/04/to-slash-or-not-to-slash.html

but looks like you got a fix for your project 👍

**Comment by dalisoft** [comment]
As i know, express exposes on both slash and without slash
